backend/uclapi/common/cachet.py

```diff
@@ -0,0 +1,127 @@
+import cachetclient.cachet as cachet
+import json
+from django.conf import settings
+
+"""
+This file handles all things cachet. From creating incidents to deleting
+them when fixed. The results can be seen on our cachet status dashboard at
+http://cachet.apps.uclapi.com/ .
+
+Incidents can be added to components of which we currently have 6 indicating
+problems with that specific system.
+List of components:
+Gencache-Staging,
+Occupeye-Staging,
+Occupeye-Mini-Staging,
+Gencache-Prod,
+Occupeye-Prod,
+Occupeye-Mini-Prod
+
+You can add a component in the admin dashboard and then reference it using
+the functions in this file without finding out its ID or attached information.
+
+More documentation on cachet endpoints can be found at
+https://docs.cachethq.io/reference
+and general documentation at
+https://docs.cachethq.io/docs
+"""
+
+
+class CachetException(Exception):
+    pass
+
+
+def get_incident_name(base):
```

Comment:
Should we add type hints? Looks like Django will never support them, but I guess it'd be nice to add them to new code that isn't Django specific. Would sphinx pick them up?If any case we use Sphinx style (it allows many different styles, right?) doc strings (for argument descriptions).